tests/test_group.py

```diff
@@ -0,0 +1,139 @@
+from time import sleep
+
+from rq import Queue, SimpleWorker
+from rq.exceptions import NoSuchGroupError
+from rq.group import Group
+from rq.job import Job
+from rq.utils import as_text
+from tests import RQTestCase
+from tests.fixtures import say_hello
+
+
+class TestGroup(RQTestCase):
+    job_1_data = Queue.prepare_data(say_hello, job_id='job1')
+    job_2_data = Queue.prepare_data(say_hello, job_id='job2')
+
+    def test_create_group(self):
+        q = Queue(connection=self.testconn)
+        group = Group.create(connection=self.testconn)
+        group.enqueue_many(q, [self.job_1_data, self.job_2_data])
+        assert isinstance(group, Group)
+        assert len(group.get_jobs()) == 2
+        q.empty
+
+    def test_group_jobs(self):
+        q = Queue(connection=self.testconn)
+        group = Group.create(connection=self.testconn)
+        jobs = group.enqueue_many(q, [self.job_1_data, self.job_2_data])
+        self.assertCountEqual(group.get_jobs(), jobs)
+        q.empty()
+
+    def test_fetch_group(self):
+        q = Queue(connection=self.testconn)
+        enqueued_group = Group.create(connection=self.testconn)
+        enqueued_group.enqueue_many(q, [self.job_1_data, self.job_2_data])
+        fetched_group = Group.fetch(enqueued_group.id, self.testconn)
+        self.assertCountEqual(enqueued_group.get_jobs(), fetched_group.get_jobs())
+        assert len(fetched_group.get_jobs()) == 2
+        q.empty()
+
+    def test_add_jobs(self):
+        q = Queue(connection=self.testconn)
+        group = Group.create(connection=self.testconn)
+        group.enqueue_many(q, [self.job_1_data, self.job_2_data])
```

Comment:
Just want to point out that this design means a single `Group` may contains jobs from multiple different queues. What are your thoughts?

Reply:
I think this is an option we'd like to have. We enqueue large groups of jobs to decrease load on an internal web API. Some of these jobs have to be on different queues because they need to be processed by different types of workers.